src/database/core/util/util.ts

```diff
@@ -114,38 +99,38 @@ export const buildLogMessage_ = function(var_args) {
  * Use this for all debug messages in Firebase.
  * @type {?function(string)}
  */
-export var logger = console.log.bind(console);
+export let logger: Function | null = console.log.bind(console);
 
 
 /**
  * Flag to check for log availability on first log message
  * @type {boolean}
  * @private
  */
-export var firstLog_ = true;
+let firstLog_ = true;
 
 
 /**
  * The implementation of Firebase.enableLogging (defined here to break dependencies)
- * @param {boolean|?function(string)} logger A flag to turn on logging, or a custom logger
- * @param {boolean=} opt_persistent Whether or not to persist logging settings across refreshes
+ * @param {boolean|?function(string)} logger_ A flag to turn on logging, or a custom logger
+ * @param {boolean=} persistent Whether or not to persist logging settings across refreshes
  */
-export const enableLogging = function(logger, opt_persistent?) {
-  assert(!opt_persistent || (logger === true || logger === false), "Can't turn on custom loggers persistently.");
-  if (logger === true) {
+export const enableLogging = function (logger_?: boolean | ((a: string) => any) | null, persistent?: boolean) {
+  assert(!persistent || (logger_ === true || logger_ === false), 'Can\'t turn on custom loggers persistently.');
+  if (logger_ === true) {
     if (typeof console !== 'undefined') {
       if (typeof console.log === 'function') {
         logger = console.log.bind(console);
       } else if (typeof console.log === 'object') {
         // IE does this.
-        logger = function(message) { console.log(message); };
+        logger = function (message) { console.log(message); };
       }
     }
-    if (opt_persistent)
+    if (persistent)
       SessionStorage.set('logging_enabled', true);
   }
-  else if (typeof logger === 'function') {
-    logger = logger;
```

Comment:
There was something weird here between the module-wide `logger` variable and the parameter passed to the `enableLoggin()` function.

Make sure to review these changes with special care.